File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/Tables/UserItem.cs
using KnowledgeSharingApi.Domains.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities.Tables
{
    [Table("UserItem")]
    public class UserItem : Entity
    {
        public Guid UserItemId { get; set; }
        public Guid UserId { get; set; }
        public virtual EUserItemType UserItemType { get; set; }

        protected override UserItem Init()
        {
            return new UserItem();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/entities/useritem.js
// Technology stack: C# Entity Framework
import Entity from "../entity";

class UserItem extends Entity {
    constructor() {
        super();
        this.UserItemId = null;
        this.UserId = null;
        this.UserItemType = null;
    }

    init() {
        return new UserItem();
    }
}

export default UserItem;